fastlane/lib/fastlane/actions/get_version_number.rb

```diff
@@ -21,6 +21,8 @@ def self.run(params)
         # Get from build settings if needed (ex: $(MARKETING_VERSION) is default in Xcode 11)
         if version_number =~ /\$\(([\w\-]+)\)/
           version_number = get_version_number_from_build_settings!(target, $1, configuration)
+        elsif version_number =~ /\$\{([\w\-]+)\}/
```

Comment:
Maybe we can combine these two regexes?
Something like `/\$(\(|\{)([\w\-]+)(\)|\})/` could work maybe.

Reply:
Yes. But no. This regex [allows](https://rubular.com/r/0xhimXyOFsgu4j) `${MARKETING_VERSION)`

The good regex is `/\$\(([\w\-]+)\)|\{([\w\-]+)\}/` but :

- it's harder to understand _(and to maintain)_
- it's requires double check to extract value from matching groups (`$1` and `$2`), so more code is finally required.

This is why I prefer an easier code than an optimized code.

Reply:
Ha, good point - you are totally right.
Maybe add a comment then for the new line a la "${MARKETING_VERSION} also works" or something like that.

Reply:
OK. Comment just append.